orttraining/orttraining/python/training/ortmodule.py

```diff
@@ -179,27 +172,29 @@ def _create_training_session(self):
             providers = ["CPUExecutionProvider"]
             provider_options = [{}]
 
-        self._forward_session = onnxruntime.InferenceSession(
-            self._onnx_forward.SerializeToString(), providers=providers, provider_options=provider_options)
-        self._backward_session = onnxruntime.InferenceSession(
-            self._onnx_backward.SerializeToString(), providers=providers, provider_options=provider_options)
+        session_options = onnxruntime.SessionOptions()
+        session_options.enable_mem_pattern = False
+        session_options.use_deterministic_compute = False
+        # 0:Verbose, 1:Info, 2:Warning. 3:Error, 4:Fatal. Default is 2.
+        session_options.log_severity_level = 1
+
+        self._gradient_session = onnxruntime.InferenceSession(
+            self._onnx_gradient.SerializeToString(), session_options, providers=providers, provider_options=provider_options)
+        
+        # Use this global run_options for now
+        self._run_options = C.RunOptions()
 
         # IO binding
         # TODO: we should try to reuse the output buffers as some of the output tensors are same sizes, expecially the backward graph outputs.
-        self._forward_io_binding = self._forward_session.io_binding()
-        self._backward_io_binding = self._backward_session.io_binding()
-
-    def _split_training_graph(self, *inputs, **kwargs):
-        # Perform shape inference and re-split forward/backward graph for batches with different shapes
-        self._module_gradient_graph_builder.build_and_split(self._current_input_shape)
-        self._onnx_forward = onnx.load_model_from_string(self._module_gradient_graph_builder.get_forward_model())
-        self._onnx_backward = onnx.load_model_from_string(self._module_gradient_graph_builder.get_backward_model())
+        self._gradient_io_binding = self._gradient_session.io_binding()
+
+    def _build_training_graph(self, *inputs, **kwargs):
+        self._module_gradient_graph_builder.build(self._current_input_shape)
```

Comment:
```suggestion
        self._module_gradient_graph_builder.build()
```
This is needed to enable dynamic axes according to @kit1980

Reply:
I tried this on NVBert, and indeed it worked. Gradient graph can build fine, and model runs, but the memory consumption is much higher. 
Let's keep the shape as input now, we need more investigation to enable this for default setting.